Tests/NIOTests/EventLoopTest.swift

```diff
@@ -228,4 +228,44 @@ public class EventLoopTest : XCTestCase {
             XCTAssertNoThrow(try group.syncShutdownGracefully())
         #endif
     }
+    
+    public func testCurrentEventLoop() throws {
+        class EventLoopHolder {
+            weak var loop: EventLoop?
+            init(_ loop: EventLoop) {
+                self.loop = loop
+            }
+        }
+        
+        func testCurrentEventLoop0() throws -> EventLoopHolder {
+            let group = MultiThreadedEventLoopGroup(numThreads: 2)
+            
+            let loop1 = group.next()
+            let currentLoop1 = try loop1.submit {
+                MultiThreadedEventLoopGroup.currentEventLoop
+            }.wait()
+            XCTAssertTrue(loop1 === currentLoop1)
+            
+            let loop2 = group.next()
+            let currentLoop2 = try loop2.submit {
+                MultiThreadedEventLoopGroup.currentEventLoop
+            }.wait()
+            XCTAssertTrue(loop2 === currentLoop2)
+            XCTAssertFalse(loop1 === loop2)
+            
+            let holder = EventLoopHolder(loop2)
+            XCTAssertNotNil(holder.loop)
+            XCTAssertNil(MultiThreadedEventLoopGroup.currentEventLoop)
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+            return holder
+        }
+        
+        let holder = try testCurrentEventLoop0()
+        
+        // Check we not leak the EventLoop.
+        while holder.loop != nil {
```

Comment:
@Lukasa I had to use a while loop here as it sometimes not directly "clear" the weak reference when `testCurrentEventLoop0` returns.